src/GeneratedHydrator/CodeGenerator/Visitor/HydratorMethodsVisitor.php

```diff
@@ -83,8 +83,8 @@ private function findAllInstanceProperties(?ReflectionClass $class = null) : arr
             $this->findAllInstanceProperties($class->getParentClass() ?: null), // of course PHP is shit.
             array_values(array_filter(
                 $class->getProperties(),
-                static function (ReflectionProperty $property) : bool {
-                    return ! $property->isStatic();
+                static function (ReflectionProperty $property) use ($class) : bool {
+                    return ! ($property->isStatic() || $property->class !== $class->name);
```

Comment:
Please avoid using public properties from `ext-reflection`: they are very unreliable.

Besides that: could you expand https://github.com/Ocramius/GeneratedHydrator/blob/ad230228c71c32279af94ed1484b907cd403886f/tests/GeneratedHydratorTest/CodeGenerator/Visitor/HydratorMethodsVisitorTest.php to verify this change?

Reply:
Refactored property access to use functions instead. Thanks for the heads up 👍 

I looked into the testing before but didn't see a nice way to test this. The easiest way would be to inspect the property maps of the HydratorMethodsVisitor. They are currently private and there are no getters for them. Would you consider weakening the isolation for them to enable easier testing or should I look into other possibilities?

Reply:
> I looked into the testing before but didn't see a nice way to test this.

Verifying the generated code is the correct way here. I can't remember how I initially tested this, but I'd suggest expanding https://github.com/Ocramius/GeneratedHydrator/blob/ad230228c71c32279af94ed1484b907cd403886f/tests/GeneratedHydratorTest/CodeGenerator/Visitor/HydratorMethodsVisitorTest.php to include an "expected generated method body" (can compare AST of produced nodes against AST of a parsed expected block)

Reply:
@Ocramius added a test to verify the change. It's quite ugly and will break if the parser changes structure of the AST but I guess that's a trade off we are willing to make.